contracts/presale/Presale.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.6.6;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../../libraries/PresaleConstants.sol";
+import "../../interfaces/IServiceHybridToken.sol";
+
+
+contract Presale is Ownable {
+    using SafeMath for uint256;
+
+    IERC20 USDC;
+    IServiceHybridToken SHBT;
+
+    uint internal constant duration = 6500 * 5; // blocks
+
+    uint internal rate;
+    uint internal startBlock;
+    uint internal totalLimit;
+    uint internal purchasedLimit;
+
+    mapping(address => uint) private purchasedAmountOf;
+
+    event Sold(address account, uint amount);
+
+    constructor(address _USDC, address _SHBT) public {
+        USDC = IERC20(_USDC);
+        SHBT = IServiceHybridToken(_SHBT);
+    }
+
+    function start() external onlyOwner {
+        startBlock = block.number;
+    }
+
+    function burn() external onlyOwner {
+        require(block.number > startBlock + duration, "Prensale: INVALID_DATE");
```

Comment:
I think that "Presale: INVALID_DATE" will do better  than "Prensale: INVALID_DATE".